src/Symfony/Component/Process/Process.php

```diff
@@ -1220,6 +1223,36 @@ public function getStartTime(): float
         return $this->starttime;
     }
 
+    /**
+     * setWinOptions sets windows proc_open specific options
+     * useful with the option create_new_console to open a process and an detach it
+     * from your php script so it can run in background
+     * requires minimum PHP 7.4.4.
+     */
+    public function setWinOptions(array $options)
```

Comment:
this should be moved as a constructor argument to me
also, no need to enforce it for windows only: some options are for linux also
and no need to validate the options: that'll make the array future-proof
makes sense to you?

Reply:
Form what I understand using php documentation of [proc_open ](https://www.php.net/manual/en/function.proc-open.php)all options are windows only, basically the `$other_options` from `proc_open` is similar to Popen creationFlags(which are also windows only), both php and python use a `STARTUPINFOW` object to pass to these flags to the native `CreateProcess`.

That's why I thought it should be a windows only method that sets these options, also is important to note that some flags may be incompatible with others for example if you use `CREATE_NEW_CONSOLE`, you can't use `DETACHED_PROCESS`( which isn't supported by PHP yet), so some basic validation should exist to avoid undesirable outcome.

I can integrate the method into the constructor if you think that makes better sense, and I can get rid of the validation checks, personally I thought is better to have a separate method setting this options since from what I can see these options will always be windows only. And you wound't want to touch these options unless you know your code might run on Windows.

Just let me know what you think and I'll try to change it that way.
Thanks.